File: gen_index.py
#! python3
# Generate the index.md file from the README.md file (just add jekyll header)

def gen_index(file: str = "README.md"):
    whole_file = None
    with open(file) as f:
        whole_file = f.read()

    with open("index.md", 'w') as f:
        f.write("---\n")
        f.write("layout: default\n")
        f.write("---\n\n")
        f.write(whole_file)

if __name__ == "__main__":
    gen_index()
            
        

File: gen_index.jl
#!/usr/bin/env julia

"Generates the index.md file from the README.md file"
function gen_index()
  whole_file = open(readstring, "README.md")

  header = """
  ---
  layout: default
  ---

  """

  open(f -> write(f, header * whole_file), "index.md", "w")
end

gen_index()
println("Generated index.md from README.md")
